### PR TITLE
Trait cleanup

### PR DESCRIPTION
**Pull request by dkrautha**
Removed the Serialize and DeserializeX traits. The implementation for primitives already didn't implement one of these traits because it needed a different function signature, and seeing serialize method calls everywhere could make it confusing as to what type is actually being operated on. There are now explicit methods for each of the sub types as well as the base type. 

Removed the boxed xdlmetadata from vec, (line 7 in xdl_vec/vec.rs). This wasn't ever needed, and just caused unnecessary allocations.

There's also some basic struct metadata implementation stuff shared in #17. That other pr should probably be merged first before this one.

### CODE REVIEW

File: xbf_rs/src/xbf_vec/vec.rs [hunk 1]
```diff
@@ -1,38 +1,13 @@
-use crate::{DeserializeType, Serialize, XbfMetadata, XbfType};
+use crate::{XbfMetadata, XbfType};
 use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
 use std::io::{self, Read, Write};
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct XbfVec {
-    pub(crate) inner_type: Box<XbfMetadata>,
+    pub(crate) inner_type: XbfMetadata,
```
Comment (ahuston-0) [inline xbf_rs/src/xbf_vec/vec.rs:7]: So we can make a XbfVec of XbfVec's without making the borrow checker mad? sweet

File: xbf_rs/tarpaulin-report.html
Comment (ahuston-0) [inline xbf_rs/tarpaulin-report.html:1]: Can we remove this from the PR? not sure if this needs to be here given that we have coveralls already taking all the coverage data

**Review by ahuston-0 — CHANGES_REQUESTED** [review]
Have one file that probably needs to get trashed and added to the gitignore but otherwise LGTM.

File: xbf_rs/src/xbf_vec/vec.rs (hunk 1 above)
Comment (dkrautha) [reply xbf_rs/src/xbf_vec/vec.rs:7]: Yup! The recursive type for metadata is XbfVecMetadata, so we need a box there, but for the XbfVec itself its not recursive with respect to that field so we can totally avoid the Box.

File: xbf_rs/tarpaulin-report.html
Comment (dkrautha) [reply xbf_rs/tarpaulin-report.html:1]: Yeah that was me making an oopsie when merging stuff in, it happened in another PR too.

**Comment by dkrautha** [comment]
Brought this up to date with main, should be good for a re-review.

**Comment by ahuston-0** [comment]
i'm still seeing the tarpaulin report. could you nuke that real quick?

**Comment by ahuston-0** [comment]
actually i can prob just nuke it gimme two seconds

**Comment by dkrautha** [comment]
> actually i can prob just nuke it gimme two seconds

there's more merge conflicts so I'll go ahead and try nuking it again anyway lol

**Comment by ahuston-0** [comment]
> > actually i can prob just nuke it gimme two seconds
> 
> there's more merge conflicts so I'll go ahead and try nuking it again anyway lol

wait pull first lol

**Comment by dkrautha** [comment]
will do

**Review by ahuston-0 — APPROVED** [review]
I feel like I approved these changes already but I guess not. Approved pending merge conflicts with main and subsequent CI

**Comment by ahuston-0** [comment]
Wait why isn't CI triggering on this branch

**Comment by ahuston-0** [comment]
looks like its doing its thing when i manually trigger it... weird. If it doesnt trigger automatically, please trigger CI manually

**Comment by dkrautha** [comment]
I think it'll trigger automatically once the conflicts are resolved, if not I'll run it manually

**Comment by ahuston-0** [comment]
woop looks good!

**Pull request merged**